### PR TITLE
CPV missing TF handling

### PR DESCRIPTION
**Pull request by sevdokim**
Hello!

In this PR missing TF protection was implemented to RawToDigitConverter.
Also I moved Hit.cxx,  CalibParams.cxx, BadChannelMap.cxx, and Pedestals.cxx to DataFormats/Detectors/CPV/.
Also I added configuration of CCDB Url to RawToDigitConverter and RawWriter.

### CODE REVIEW

File: DataFormats/Detectors/CPV/src/DataFormatsCPVLinkDef.h [hunk 1]
```diff
@@ -26,4 +28,12 @@
 #pragma link C++ struct o2::cpv::CTF + ;
 #pragma link C++ class o2::ctf::EncodedBlocks < o2::cpv::CTFHeader, 7, uint32_t> + ;
 
+#pragma link C++ class o2::cpv::BadChannelMap + ;
+#pragma link C++ class o2::TObjectWrapper < o2::cpv::BadChannelMap> + ;
```
Comment (shahor02) [inline DataFormats/Detectors/CPV/src/DataFormatsCPVLinkDef.h]: TObjectWrapper is obsolete, please remove all these entries.

File: Detectors/CPV/workflow/src/RawToDigitConverterSpec.cxx [hunk 2]
```diff
@@ -19,115 +19,135 @@
...
+    kNO_CPVTRAILER,
+    kCPVHEADER_INVALID,
+    kCPVTRAILER_INVALID,
+    kSEGMENT_HEADER_ERROR,
+    kROW_HEADER_ERROR,
+    kEOE_HEADER_ERROR,
+    kPADERROR,
+    kUNKNOWN_WORD,
+    kPadAddress
+    };*/
```
Comment (shahor02) [inline Detectors/CPV/workflow/src/RawToDigitConverterSpec.cxx]: can this be removed?

File: Detectors/CPV/workflow/src/cpv-reco-workflow.cxx [hunk 3]
```diff
@@ -62,5 +64,7 @@ o2::framework::WorkflowSpec defineDataProcessing(o2::framework::ConfigContext co
                                              !cfgc.options().get<bool>("disable-mc"),       //
                                              cfgc.options().get<std::string>("input-type"), //
                                              cfgc.options().get<std::string>("output-type") //
+ //                                            cfgc.options().get<bool>("pedestal"),
+ //                                            cfgc.options().get<std::string>("ccdb-url"),
```
Comment (shahor02) [inline Detectors/CPV/workflow/src/cpv-reco-workflow.cxx]: why do you need to add options which are not needed?

**Review by shahor02 — COMMENTED** [review]
@sevdokim thanks, see a few comments below! 
The formatting needs to be fixed. In general, it is better to avoid pushing so many unrelated changes in a single commit.

Since the FullCI will likely fail due to internal reasons and you are touching raw data handling, could you run yourself the FST (you need to have the AEGIS installed):
`$O2_ROOT/prodtests/full_system_test.sh` ?

File: Detectors/CPV/workflow/src/cpv-reco-workflow.cxx (hunk 3 above)
Comment (sevdokim) [reply Detectors/CPV/workflow/src/cpv-reco-workflow.cxx]: hello @shahor02 
they are not needed. I removed them.

File: Detectors/CPV/workflow/src/RawToDigitConverterSpec.cxx (hunk 2 above)
Comment (sevdokim) [reply Detectors/CPV/workflow/src/RawToDigitConverterSpec.cxx]: Yes, done

File: DataFormats/Detectors/CPV/src/DataFormatsCPVLinkDef.h (hunk 1 above)
Comment (sevdokim) [reply DataFormats/Detectors/CPV/src/DataFormatsCPVLinkDef.h]: Ok, thanks. Removed.

File: Detectors/CPV/workflow/src/cpv-reco-workflow.cxx (hunk 3 above)
Comment (sevdokim) [reply Detectors/CPV/workflow/src/cpv-reco-workflow.cxx]: Dear @shahor02 ,
I'll try to run full system test and let you know the result.
I also want to be sure that protection against missing TFs is working.
We can simulate missing TFs by command 
`o2-raw-file-reader-workflow --input-conf raw/CPV/CPVraw.cfg --loop 100 --drop-tf "CPV,3"`
But what I need to do next to see if everything is working as expected?

Thank you & regards,
Sergey

Comment (shahor02) [reply Detectors/CPV/workflow/src/cpv-reco-workflow.cxx]: Hi @sevdokim 
you should see in every 3d TF the message you put: `"Sending empty output due to data type input with 0xDEADBEEF"`

Comment (sevdokim) [reply Detectors/CPV/workflow/src/cpv-reco-workflow.cxx]: Dear @shahor02 ,

I tried to run FST using following environment on ubuntu20.04:
`alienv enter O2/latest AEGIS/latest`
Unfortunately the test was failed at 2nd task:
```
Launching task: o2-sim --seed -1 -n 10 --configKeyValues Diamond.width[2]=6. -g pythia8hi -e TGeant4 -j 4 &> sim.log &
command o2-sim --seed -1 -n 10 --configKeyValues Diamond.width[2]=6. -g pythia8hi -e TGeant4 -j 4 had nonzero exit code 1

```
From the sim.log I see following:
`Spawning particle server on PID 1270555; Redirect output to o2sim_serverlog`
`[INFO] Process 1270555 EXITED WITH CODE 0 SIGNALED 1 SIGNAL 6`
from the o2sim_serverlog:
` PYTHIA Error in BeamRemnants::setKinematics: kinematics construction failed  `
`[ESC[01;36m17:11:29ESC[0m][ESC[01;33mWARNESC[0m] Particle to be tracked is not defined in PDG: pdg = 1000200489`
`[ESC[01;36m17:11:29ESC[0m][ESC[01;33mWARNESC[0m] Particle to be tracked is not defined in PDG: pdg = 1000180489`
I suppose the error is not related to CPV but to non-recognition of some ions by the VMC?
I bet that my setup is not well defined and it was a cause of this failure. Any tips how to proceed?

Thank you!
Sergey

Comment (shahor02) [reply Detectors/CPV/workflow/src/cpv-reco-workflow.cxx]: Hi @sevdokim 
Will run myself to check, we recently switched to G4, might be related to this.

**Comment by shahor02** [comment]
@sevdokim The FST seems to be broken at the moment, I've tested your branch on existing raw data and had to apply some changes to make lost TF recovery to work. Please merge the https://github.com/sevdokim/AliceO2/pull/2 I opened on your repo, then I will merge this PR w/o waiting again for the tests.

File: Detectors/CPV/workflow/src/cpv-reco-workflow.cxx (hunk 3 above)
Comment (sevdokim) [reply Detectors/CPV/workflow/src/cpv-reco-workflow.cxx]: hi @shahor02 ,
done from my side! thanks!

**Review by shahor02 — APPROVED** [review]
Merging as the tests were passed before and last fix was tested locally.

**Pull request merged**